lib/imgix/path.rb

```diff
@@ -11,44 +11,12 @@ def initialize(prefix, secure_url_token, path = "/")
       @prefix = prefix
       @secure_url_token = secure_url_token
       #TODO(luqven): should this method not be invoked on initialization?
-      @path = sanitize_path(path)
+      @path = path
       @options = {}
     end
 
-    # Escape and encode any characters in path that are reserved and not utf8 encoded.
-    # This includes " +?:#" characters. If a path is being used as a proxy, utf8
-    # encode everything. If it is not being used as proxy, leave certain chars, like
-    # "/", alone. Method assumes path is not already encoded.
-    def sanitize_path(path)
-      # remove the leading "/", we'll add it back after encoding
-      path = path.slice(1, path.length) if Regexp.new('^/') =~ path
-      # if path is being used as a proxy, encode the entire thing
-      if /^https?/ =~ path
-        return encode_URI_Component(path) 
-      else
-        # otherwise, encode only specific characters
-        return encode_URI(path)
-      end
-    end
-
-    # URL encode the entire path
-    def encode_URI_Component(path)
-      return "/" + CGI.escape(path)
-    end
-
-    # URL encode every character in the path, including
-    # " +?:#" characters.
-    def encode_URI(path)
-      result = []
-      path.split("/").each do |str|
-        escaped_key = ERB::Util.url_encode(str)
-        result << escaped_key
-      end
-      result =  "/" + result.join("/")
-      return result;
-    end
-
     def to_url(opts = {})
+      @path = sanitize_path(@path)
```

Comment:
👍 👍

Reply:
Will this cause multiple encoding issues if `to_url` is called multiple times on the same `Path`? Perhaps a different ivar name and `||= sanitize_path(@path)` could be used instead.

Reply:
Great suggestion 💯 , I'll try this out locally and push the change if all looks good.